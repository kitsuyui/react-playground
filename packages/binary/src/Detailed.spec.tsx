import { describe, it, expect } from 'vitest'

import { render } from '@testing-library/react'

import {
  DetailedBit,
  DetailedBitString
} from './Detailed'


describe('DetailedBitString component', () => {
  it('should render properly with 8 bits', () => {
    const value = 0b10101010
    const { asFragment } = render(<DetailedBitString value={value} />)
    expect(asFragment()).toMatchSnapshot()
  })
})

describe('DetailedBit component', () => {
  it('should render "1" for stand=true and overflow=false', () => {
    const { asFragment } = render(<DetailedBit stand={true} overflow={false} littleEndianBitOrder={0} bigEndianBitOrder={0} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders overflow with red dotted border', () => {
    const { asFragment } = render(<DetailedBit stand={true} overflow={true} littleEndianBitOrder={0} bigEndianBitOrder={0} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders zero padding with gray dotted border', () => {
    const { asFragment } = render(<DetailedBit stand={false} overflow={false} zeroPadding={true} littleEndianBitOrder={0} bigEndianBitOrder={0} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders with little endian bit order', () => {
    const { asFragment } = render(<DetailedBit stand={true} overflow={false} littleEndianBitOrder={1} bigEndianBitOrder={0} displayBigEndianBitOrder={true} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders with big endian bit order', () => {
    const { asFragment } = render(<DetailedBit stand={true} overflow={false} littleEndianBitOrder={0} bigEndianBitOrder={1} displayLittleEndianBitOrder={true} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
