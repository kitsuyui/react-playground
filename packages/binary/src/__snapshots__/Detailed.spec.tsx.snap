// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DetailedBit component > renders overflow with red dotted border 1`] = `
<DocumentFragment>
  <div
    style="display: inline-block; font-family: monospace; margin: 0.5px;"
  >
    <div
      style="color: rgb(153, 153, 153); background-color: rgb(0, 0, 0); border: 1px dotted red;"
    >
      1
    </div>
  </div>
</DocumentFragment>
`;

exports[`DetailedBit component > renders with big endian bit order 1`] = `
<DocumentFragment>
  <div
    style="display: inline-block; font-family: monospace; margin: 0.5px;"
  >
    <div
      style="color: rgb(153, 153, 153); background-color: rgb(0, 0, 0); border: 1px solid rgb(204, 204, 204);"
    >
      1
    </div>
    <div
      style="user-select: none; font-size: 0.5em; text-align: center; line-height: 1em;"
    >
      0
    </div>
  </div>
</DocumentFragment>
`;

exports[`DetailedBit component > renders with little endian bit order 1`] = `
<DocumentFragment>
  <div
    style="display: inline-block; font-family: monospace; margin: 0.5px;"
  >
    <div
      style="color: rgb(153, 153, 153); background-color: rgb(0, 0, 0); border: 1px solid rgb(204, 204, 204);"
    >
      1
    </div>
    <div
      style="user-select: none; font-size: 0.5em; text-align: center; line-height: 1em;"
    >
      0
    </div>
  </div>
</DocumentFragment>
`;

exports[`DetailedBit component > renders zero padding with gray dotted border 1`] = `
<DocumentFragment>
  <div
    style="display: inline-block; font-family: monospace; margin: 0.5px;"
  >
    <div
      style="color: rgb(204, 204, 204); background-color: rgb(255, 255, 255); border: 1px dotted gray;"
    >
      0
    </div>
  </div>
</DocumentFragment>
`;

exports[`DetailedBit component > should render "1" for stand=true and overflow=false 1`] = `
<DocumentFragment>
  <div
    style="display: inline-block; font-family: monospace; margin: 0.5px;"
  >
    <div
      style="color: rgb(153, 153, 153); background-color: rgb(0, 0, 0); border: 1px solid rgb(204, 204, 204);"
    >
      1
    </div>
  </div>
</DocumentFragment>
`;

exports[`DetailedBitString component > should render properly with 8 bits 1`] = `
<DocumentFragment>
  <div
    style="display: inline-block; font-family: monospace; margin: 0.5px;"
  >
    <div
      style="color: rgb(153, 153, 153); background-color: rgb(0, 0, 0); border: 1px solid rgb(204, 204, 204);"
    >
      1
    </div>
  </div>
  <div
    style="display: inline-block; font-family: monospace; margin: 0.5px;"
  >
    <div
      style="color: rgb(204, 204, 204); background-color: rgb(255, 255, 255); border: 1px solid rgb(204, 204, 204);"
    >
      0
    </div>
  </div>
  <div
    style="display: inline-block; font-family: monospace; margin: 0.5px;"
  >
    <div
      style="color: rgb(153, 153, 153); background-color: rgb(0, 0, 0); border: 1px solid rgb(204, 204, 204);"
    >
      1
    </div>
  </div>
  <div
    style="display: inline-block; font-family: monospace; margin: 0.5px;"
  >
    <div
      style="color: rgb(204, 204, 204); background-color: rgb(255, 255, 255); border: 1px solid rgb(204, 204, 204);"
    >
      0
    </div>
  </div>
  <div
    style="display: inline-block; font-family: monospace; margin: 0.5px;"
  >
    <div
      style="color: rgb(153, 153, 153); background-color: rgb(0, 0, 0); border: 1px solid rgb(204, 204, 204);"
    >
      1
    </div>
  </div>
  <div
    style="display: inline-block; font-family: monospace; margin: 0.5px;"
  >
    <div
      style="color: rgb(204, 204, 204); background-color: rgb(255, 255, 255); border: 1px solid rgb(204, 204, 204);"
    >
      0
    </div>
  </div>
  <div
    style="display: inline-block; font-family: monospace; margin: 0.5px;"
  >
    <div
      style="color: rgb(153, 153, 153); background-color: rgb(0, 0, 0); border: 1px solid rgb(204, 204, 204);"
    >
      1
    </div>
  </div>
  <div
    style="display: inline-block; font-family: monospace; margin: 0.5px;"
  >
    <div
      style="color: rgb(204, 204, 204); background-color: rgb(255, 255, 255); border: 1px solid rgb(204, 204, 204);"
    >
      0
    </div>
  </div>
</DocumentFragment>
`;
